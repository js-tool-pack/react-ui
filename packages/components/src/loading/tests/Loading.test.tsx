import BasicDemo from '../demo/basic';
import GlobalDemo from '../demo/global';
import HolderDemo from '../demo/useLoadingHolder';
import { render, fireEvent } from '@testing-library/react';

describe('Loading', () => {
  test('basic', async () => {
    const { container } = render(<BasicDemo />);
    expect(document.body.querySelector('.t-loading__ref')).toMatchSnapshot();
    fireEvent.click(container.querySelector('.t-button')!);
    expect(document.body.querySelector('.t-loading__ref')).toMatchSnapshot();
  });

  test('global', async () => {
    const { container } = render(<GlobalDemo />);
    const getLoading = () => document.body.querySelector('.t-loading');
    expect(getLoading()).toBeNull();
    fireEvent.click(container.querySelector('.t-button')!);
    expect(getLoading()).not.toBeNull();
  });

  test('holder', async () => {
    const { container } = render(<HolderDemo />);
    const getLoading = () => document.body.querySelector('.t-loading');
    expect(getLoading()).toBeNull();
    fireEvent.click(container.querySelector('.t-button')!);
    expect(getLoading()).not.toBeNull();
  });
});
