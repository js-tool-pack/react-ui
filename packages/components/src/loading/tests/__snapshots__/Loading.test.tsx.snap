// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading basic 1`] = `
<div
  class="t-loading__ref"
  style="margin-top: 20px; height: 100px; background: lime; line-height: 100px;"
>
  我是内容
</div>
`;

exports[`Loading basic 2`] = `
<div
  class="t-loading__ref"
  style="margin-top: 20px; height: 100px; background: lime; line-height: 100px;"
>
  我是内容
  <div
    class="t-loading t-loading-enter-active t-loading-enter-to"
    style="z-index: 1000;"
  >
    <div
      class="t-loading__box"
    >
      <div
        class="t-loading__icon"
      >
        <i
          class="t-icon"
          role="img"
          style="font-size: 38px;"
        >
          <svg
            class="t-loading-icon"
            viewBox="0 0 66 66"
          >
            <circle
              class="t-loading-icon__path"
              cx="33"
              cy="33"
              fill="none"
              r="30"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="6"
            />
          </svg>
        </i>
      </div>
      <div
        class="t-loading__text"
      >
        loading...
      </div>
    </div>
  </div>
</div>
`;

exports[`Loading global 1`] = `
<body>
  <div>
    <div
      style="text-align: center;"
    >
      <button
        class="t-button t-button--type-primary t--size-m"
      >
        <span>
          显示
          全屏的loading
        </span>
      </button>
    </div>
  </div>
</body>
`;

exports[`Loading global 2`] = `
<body>
  <div>
    <div
      style="text-align: center;"
    >
      <button
        class="t-button t-button--type-primary t--size-m"
      >
        <span>
          隐藏
          全屏的loading
        </span>
        <span
          class="t-button__wave t-button__wave--active"
        />
      </button>
    </div>
  </div>
  <div
    class="t-loading t-loading-enter-active t-loading-enter-to"
    style="z-index: 1000;"
  >
    <div
      class="t-loading__box"
    >
      <div
        class="t-loading__icon"
      >
        <i
          class="t-icon"
          role="img"
          style="font-size: 38px;"
        >
          <svg
            class="t-loading-icon"
            viewBox="0 0 66 66"
          >
            <circle
              class="t-loading-icon__path"
              cx="33"
              cy="33"
              fill="none"
              r="30"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="6"
            />
          </svg>
        </i>
      </div>
      <div
        class="t-loading__text"
      >
        loading...
      </div>
    </div>
  </div>
</body>
`;
